packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -19,37 +22,79 @@ import 'scrollable.dart';
 import 'sliver.dart';
 import 'viewport.dart';
 
+/// A controller for [PageView].
+///
+/// A page controller lets you manipulate which page is visible in a [PageView].
+///
+/// See also:
+///
+///  - [PageView], which is the widget this object controls.
 class PageController extends ScrollController {
+  /// Creates a page controller.
+  ///
+  /// The [initialPage] and [peek] arguments must not be null.
   PageController({
     this.initialPage: 0,
+    this.peek: 0.0,
   }) {
     assert(initialPage != null);
+    assert(peek != null);
   }
 
+  /// The page to show when first creating the [PageView].
   final int initialPage;
 
+  /// The number of pixels of the next and previous page that should be visible
+  /// before and after the current page.
+  final double peek;
```

Comment:
Another way to define this would have been been pageWidth as a percentage of the viewport's width. The default and the maximum would be 1.0.